src/cpp/src/sampling/logit_processor.hpp

```diff
@@ -332,9 +45,17 @@ class LogitProcessor {
             );
         }
 
+        #ifdef ENABLE_XGRAMMAR
+        OPENVINO_ASSERT(structured_output_controller != nullptr, "Structured output controller is not initialized");
```

Comment:
What do you think if we incapsulate this into `StructuredOutputController`?

Reply:
Is it possible to move all `ifdef ENABLE_XGRAMMAR` into `StructuredOutputController`? And preserve `StructuredOutputController` unconditionally?